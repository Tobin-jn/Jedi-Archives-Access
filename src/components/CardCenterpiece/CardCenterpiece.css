.card-centerpiece {
  grid-column: 2 / 4;
  grid-row: 2;
  /*border: solid rgba(173,216,230,0.2);*/
  height: 250px;
  width: 500px;
  display: grid;
  justify-content: center;
}

.centerpiece-icon {
  width: 140px;
  text-align: center;
  font-size: 5em;
  color: #ADD8E6;
  opacity: 1;
  margin: 1rem auto;
}

.centerpiece-text {
  font-family: Andale Mono,AndaleMono,monospace;
  color: #ADD8E6;
  font-size: 1.7em;
  text-align: center;
  padding: 20px;
  text-transform: uppercase;
  margin-top: 0;
  /*opacity: 0.7;*/
}

@media (max-width: 1000px) {
  .card-centerpiece {
    grid-column: 2 / 3;
    width: 250px;
  }
}

@media (max-width: 750px) {
  .card-centerpiece {
    grid-column: 1 / 3;
    width: 500px;
  }
}

@media (max-width: 500px) {
  .card-centerpiece {
    grid-column: 1;
    width: 250px;
  }
}