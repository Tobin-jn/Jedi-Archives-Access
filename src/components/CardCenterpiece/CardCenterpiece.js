import React from 'react';
import './CardCenterpiece.css';
import PropTypes from'prop-types';

const CardCenterpiece = ({selection}) => {
  let centerpiece;

  // if (selection === 'people'){
  //   centerpiece = <div>People</div>
  // } else if (selection === 'planets') {
  //   centerpiece = <p>planets</p>
  // } else if (selection === 'vehicles') {
  //   centerpiece = <p>vehicles</p>
  // } else if (selection === 'favorites') {
  //   centerpiece = <p>favorites</p>
  // } 
      // {centerpiece}

  return(
    <div className="card-centerpiece">
      <i className='fab fa-jedi-order centerpiece-icon'></i>
      <h3 className='centerpiece-text'>Persons of Interest</h3>
    </div>
  )
}

export default CardCenterpiece