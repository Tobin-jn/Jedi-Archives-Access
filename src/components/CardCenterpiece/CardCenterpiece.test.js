/* eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import CardCenterpiece from './CardCenterpiece';


describe('CardCenterpiece', () => {
  let wrapper;
  let mockSelection;


  it('should exist', () => {
    mockSelection = 'people';
    wrapper = shallow(<CardCenterpiece totalFavorites={3} selection={mockSelection} />);

    expect(wrapper).toBeDefined();
  });

  it('should match snapshot if selection is people', () => {
    mockSelection = 'people';
    wrapper = shallow(<CardCenterpiece totalFavorites={3} selection={mockSelection} />);
    
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if selection is planets', () => {
    mockSelection = 'planets';
    wrapper = shallow(<CardCenterpiece totalFavorites={3} selection={mockSelection} />);
    
    expect(wrapper).toMatchSnapshot();
  });

    it('should match snapshot if selection is vehicles', () => {
    mockSelection = 'vehicles';
    wrapper = shallow(<CardCenterpiece totalFavorites={3} selection={mockSelection} />);
    
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if selection is favorites', () => {
    mockSelection = 'favorites';
    wrapper = shallow(<CardCenterpiece totalFavorites={3} selection={mockSelection} />);
    
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if selection is not people/planets/vehicles/favorites', () => {
    mockSelection = 'films';
    wrapper = shallow(<CardCenterpiece totalFavorites={3} selection={mockSelection} />);
    
    expect(wrapper).toMatchSnapshot();
  });
})