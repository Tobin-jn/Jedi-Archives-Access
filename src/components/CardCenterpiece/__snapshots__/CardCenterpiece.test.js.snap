// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCenterpiece should match snapshot if selection is favorites 1`] = `
<div
  className="card-centerpiece"
>
  <i
    className="fab fa-jedi-order centerpiece-icon"
  />
  <h3
    className="centerpiece-text"
  >
    Saved items
  </h3>
  <p
    className="centerpiece-fav-text"
  >
    tap to remove data
  </p>
</div>
`;

exports[`CardCenterpiece should match snapshot if selection is not people/planets/vehicles/favorites 1`] = `
<div
  className="card-centerpiece"
>
  <i
    className="fab fa-jedi-order centerpiece-icon"
  />
  <h3
    className="centerpiece-text"
  />
  <p
    className="centerpiece-fav-text"
  >
    tap to save data
  </p>
</div>
`;

exports[`CardCenterpiece should match snapshot if selection is people 1`] = `
<div
  className="card-centerpiece"
>
  <i
    className="fab fa-jedi-order centerpiece-icon"
  />
  <h3
    className="centerpiece-text"
  >
    Persons of Interest
  </h3>
  <p
    className="centerpiece-fav-text"
  >
    tap to save data
  </p>
</div>
`;

exports[`CardCenterpiece should match snapshot if selection is planets 1`] = `
<div
  className="card-centerpiece"
>
  <i
    className="fab fa-jedi-order centerpiece-icon"
  />
  <h3
    className="centerpiece-text"
  >
    Locations of Interest
  </h3>
  <p
    className="centerpiece-fav-text"
  >
    tap to save data
  </p>
</div>
`;

exports[`CardCenterpiece should match snapshot if selection is vehicles 1`] = `
<div
  className="card-centerpiece"
>
  <i
    className="fab fa-jedi-order centerpiece-icon"
  />
  <h3
    className="centerpiece-text"
  >
    Available Conveyance
  </h3>
  <p
    className="centerpiece-fav-text"
  >
    tap to save data
  </p>
</div>
`;
